src/main/java/com/squareup/javapoet/CodeBlock.java

```diff
@@ -269,6 +278,13 @@ public Builder addStatement(String format, Object... args) {
       return this;
     }
 
+    public Builder addStatement(CodeBlock codeBlock) {
```

Comment:
Sorry for the late response. It's not really important for me whether we include that function or not, so if it's necessary I'll remove it. 

A few notes about those lines:
- It looks much better: `builder.addStatement(codeBlock)` vs `codeBlockBuilder.add(";\n")` and then `builder.addCode(codeBlock)`
- It gives a system more information about our code (I'm talking about `add("$[")` and `add(";\n$]")`